docs/CODE_REVIEW.md

```
@@ -1,306 +1,299 @@
+## Architectural Patterns
 
-### Props Interface Naming
-```typescript
-// ✅ Component-specific
-interface ChatHeaderProps { ... }
-export interface ChatPanelHandle { focusInput: () => void; }
+### 1. Feature-Based Directory Structure
 
-// ❌ Generic
-interface Props { ... }
 ```
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)